oneview_redfish_toolkit/api/computer_system.py

```diff
@@ -115,3 +143,39 @@ def map_boot(self, boot_list):
             redfish_boot_list.append('None')
 
         return redfish_boot_list
+
+    def get_oneview_power_configuration(self, reset_type):
+        """Maps Redfish's power options to OneView's power option
+
+            Maps the known Redfish power options to OneView Power option.
+            If a unknown power option shows up it will raise an Exception.
+
+            Args:
+                reset_type: Redfish power option.
+
+            Returns:
+                dict: Dict with OneView power configuration.
+
+            Exception:
+                OneViewRedfishError: raises an exception if
+                reset_type is an unmapped value.
+        """
+
+        if reset_type == "PushPowerButton":
+            if self.server_hardware["powerState"] == "On":
+                POWER_STATE_MAP["PushPowerButton"]["powerState"] = "Off"
+            else:
+                POWER_STATE_MAP["PushPowerButton"]["powerState"] = "On"
+
+        if reset_type == "ForceOn" or reset_type == "Nmi":
```

Comment:
Test abort conditions first, If you are going to abort why did you run the code above?!?

Reply:
Fixed.